### ISSUE TITLE
[MHV Horizon / Accessibility] Account Activity Logs AI prototype

### ISSUE THREAD

**Issue by sterkenburgsara** [department-of-veterans-affairs/va.gov-team#134312]
### 📝 Description

# Background
Building a new page in the My HealtheVet portal for Account activity logs. 

### 📝 Background and context

# Background
Hello, the Horizon team is working on a "lift and shift" project to move a feature called "Account Activity Logs" from a legacy health care patient portal known as My HealtheVet Classic over to MHV on VA.gov. This is not a high demand feature, but is both a HIPAA requirement and is sort of the "long pole in the tent" - the last remaining feature that needs to move over before we can sunset that legacy portal come September. 

## Design plan & goals
The AALs will be a single page built at `/my-health/account-activity` and will give users an audit trail of the health tools and events that took place or were reviewed in their health portal over time. Right now, we are thinking that it will need to have lightweight functionality to narrow the list, including a filter and sort component with 3 filter facets. 

Each event will have a card, and because the data and information is so limited for AALs, we do not believe that details pages are necessary or beneficial with this feature. Instead, our plan is to put little information in the card, and have additional details in an additional info component that can be expanded/collapsed. 

## Desired outcomes & using AI
Our goal is to use this feature to test out the new VADS AI prototyping kit and MCP server to see if we can generate an accessible, working prototype that leverages VADS components. This use-case will require some complexity that is not generally supported yet by the prototyping kit, but Matt Dingee will work with our team and come to UX syncs when we're far enough along to help us through this as needed. I am wondering if accessibility can also help me with the following: 
* A couple of components in the health portal are not supported centrally by VADS, as you know. The ones that I'd like to put in this prototype that are not supported include: 
  * MHV secondary navigation bar
  * Your improved filter / sort component  

I will need help generating these with AI and evaluating the code output to make sure it aligns with our current coded implementation and that it stays accessible. I'd love to partner with you as I dive down this rabbit hole and explore how to do this! 

## Other challenges
* Structured JSON for AAL events is something I need to create w/ our engineer for this work, Kevin Duensing - _in process_
* Test user data to mock realistic cards - _in process_

### ✋ Point of contact

Sara Sterkenburg

### 🛠️ Product

Account Activity Logs

### 🗣️ Slack channel

#mhv-horizon

### ⏳ Timeframe

This is a fast-moving project and will not go to research. We are still deciding whether or not to do all collaboration cycle touchpoints or not. Staging review timeline is not firm, but estimated April-May. 

### Evaluation Support

- [ ] Design review — Review design artifacts to identify accessibility issues and recommend solutions.
- [ ] Conformance testing — Perform an accessibility audit of a functional user flow to identify accessibility issues and recommend solutions.
- [ ] Accessibility artifact— Prepare an accessibility artifact for a Collaboration Cycle review and identify issues and recommend solutions.

### Implementation Support

- [ ] Design — Prepare accessible designs, collaborate on design tasks and resolving tickets, etc.
- [ ] Engineering — Prepare accessible prototypes, collaborate on engineering tasks and resolving tickets, etc.

### General Support

- [ ] Issue review — Review tickets or other reports of accessibility issues and provide guidance on how to address issues.
- [x] Consulting — Provide general guidance and training on accessibility topics.

### 🔗 Supporting resources

- Figma file: https://www.figma.com/design/Jr5ytaVMIsYn9L283bq35y/Account-Activity-Logs--AAL-?node-id=124-3
- [VA prototype kit & instructions](https://github.com/department-of-veterans-affairs/va-prototype-kit)
- [VADS MCP server](https://github.com/department-of-veterans-affairs/vads-mcp-server)
- [Account activity page created with VADS prototyping kit](https://department-of-veterans-affairs.github.io/va-prototype-kit/src/prototypes/account-activity/index.html)

#### General: Refine the ticket and work
- [x] Add ticket labels — Add the product's project and labels.
- [x] Save and submit — Save and submit this intake ticket.
- [x] Share on Slack — Start a new thread and post a link to this ticket in the mhv-accessibility Slack channel — see [Slack thread](https://dsva.slack.com/archives/C05QG2H510T/p1771969892879939)
- [ ] Refine work and revise ticket

#### Workstream 1: Optimize prototype output for use in research with AT users (lead, @sarahhortonadhoc)
- [x] Evaluate output using staging review accessibility artifact — #136189
- [x] Evaluate output using MHV WCAG conformance audit process — See [conformance audit worksheet (Google)](https://docs.google.com/spreadsheets/d/11HE2B-PnisZ7xAKMzIoqbQ9HFmsyTrVFmXmTinA7bBc/edit?pli=1&gid=1007863918#gid=1007863918&range=B:B)
- [ ] Document accessibility issues in the prototype kit output
- [ ] Draft proposal for mitigating potential accessibility issues produced by prototype kit

#### Workstream 2: Include core MHV components in prototyping kit (lead, @steveg-IAT)
- [ ] Identify priority components/patterns (Secondary nav, Filter/sort, Print/download)
- [ ] TBC

#### Workstream 3: Develop processes for incorporating prototypes into development lifecycle (lead, @sterkenburgsara)
- [ ] TBC

**Comment by sarahhortonadhoc** [comment]
### Additional insights from prototype kit output

#### Start a new message research prototype — Apr 2026

- **Error messages** — inputs have error messages that use the role attribute to provide the error announcement to screen reader users, e.g.,`<span id="input-error-message" role="alert"></span>`. In the prototype, the element is announced in reading mode, even when there is no error, as "alert". It seems like there are elements that are part of patterns that have the role attribute. Since they have the role attribute they are announced in reading mode, e.g., `<span class="loading-message" role="status"></span>`, even when there is no content (opposite issue to dynamic alerts, below, which need the `role` attribute to be implemented correctly).
- **Focus management** 
  - Focus is set to the top of the page on page load rather than the page heading. The convention for MHV is to focus on the page heading — see [Focus Management on Page Load](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/accessibility/documentation/decision-records.md#focus-management-on-page-load). This also applies to the message sent page — focus should be set to the page heading.
  - Focus should be set to the input in error when an error message displays — see [VADS Form guidance](https://design.va.gov/components/form/#what-to-do-instead). Verify this is happening for all of the different input types, e.g., it doesn't appear to be working as intended for the combobox.
- **Alerts** — dynamic alerts need the `role` attribute to be announced — see [dynamic alerts pattern documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/accessibility/documentation/patterns.md#notification).

**Comment by jeana-adhoc** [comment]
@sarahhortonadhoc - I believe the first date issue you mentioned is a known issue. [We're working on a PR for it here](https://github.com/department-of-veterans-affairs/component-library/pull/1962#pullrequestreview-4007885322)
